File: Team_Workspace/dalia_ahmed/task9.py

#parent
class Vehicle:
    def __init__(self):
        self._speed=0.0
    def set_speed(self, s):
        self._speed=s
    def show_speed(self):
        print("current speed:",self._speed)


class SportsCar(Vehicle):
    def turbo_boost(self):
        self._speed*=2
        print("TURBO ACTIVATED!")




my_car=SportsCar()
my_car.set_speed(50)
my_car.turbo_boost()
my_car.show_speed()

File: Team_Workspace/Fatma_Ahmed/cpp_tasks/task9/task9.cpp
//task9

#include <iostream>
using namespace std;

class Vehicle {
protected:
    double speed;
public:
    void setSpeed(double s)
    {
        this->speed = s;

    }
    void showSpeed() {
        cout << "Current Speed: " << speed<<endl;

    }
};

class SportsCar :public Vehicle {
public:
    void turboBoost() {
        this->speed *= 2;
        cout << "TURBO ACTIVATED\n";

    };

};
int main()
{
    SportsCar TestCar;
    TestCar.setSpeed(50);
    TestCar.turboBoost();
    TestCar.showSpeed();

}

